src/System.Memory/src/System/Buffers/IRetainable.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Provides a mechanism to make sure objects only get disposed when intended.
```

Comment:
I would say something along the lines of "Interface representing objects with manual lifetime management"